redecs_agent/redecs_agent.go

```diff
@@ -20,6 +20,95 @@ import (
 	"time"
 )
 
+const workerTimeout = 180 * time.Second
+
+type handler interface {
+	Handle(*docker.APIEvents) error
+}
+
+type dockerRouter struct {
+	handlers      map[string][]handler
+	dockerClient  *docker.Client
+	listener      chan *docker.APIEvents
+	workers       chan *worker
+	workerTimeout time.Duration
+}
+
+func dockerEventsRouter(bufferSize int, workerPoolSize int, dockerClient *docker.Client,
```

Comment:
dockerRouter can be moved into its own file or subpackage (everything within a folder is in the same package)